ShopAppTests/App/Account/Controllers/AddressFormViewControllerSpec.swift

```diff
@@ -0,0 +1,277 @@
+                    expect(viewController.viewModel.lastNameText.value) == "last name"
+                    expect(viewController.viewModel.addressText.value) == "address"
+                    expect(viewController.viewModel.addressOptionalText.value) == "second address"
+                    expect(viewController.viewModel.cityText.value) == "city"
+                    expect(viewController.viewModel.stateText.value) == "state"
+                    expect(viewController.viewModel.zipText.value) == "zip"
+                    expect(viewController.viewModel.phoneText.value) == "phone"
+                }
+            }
+        }
+        
+        describe("when text field views text changed") {
+            beforeEach {
+                viewController = UIStoryboard(name: StoryboardNames.account, bundle: nil).instantiateViewController(withIdentifier: ControllerIdentifiers.addressForm) as! AddressFormViewController
+                
+                let repositoryMock = PaymentsRepositoryMock()
+                let addressFormUseCaseMock = CountriesUseCase(repository: repositoryMock)
+                viewModelMock = AddressFormViewModelMock(countriesUseCase: addressFormUseCaseMock)
+                viewController.viewModel = viewModelMock
+                
+                countryPicker = self.findView(withAccessibilityLabel: "addressFormCountryPicker", in: viewController.view) as! BasePicker
+                nameTextFieldView = self.findView(withAccessibilityLabel: "addressFormName", in: viewController.view) as! InputTextFieldView
+                lastNameTextFieldView = self.findView(withAccessibilityLabel: "addressFormLastName", in: viewController.view) as! InputTextFieldView
+                addressTextFieldView = self.findView(withAccessibilityLabel: "addressFormAddress", in: viewController.view) as! InputTextFieldView
+                addressOptionalTextFieldView = self.findView(withAccessibilityLabel: "addressFormAddressOptional", in: viewController.view) as! InputTextFieldView
+                cityTextFieldView = self.findView(withAccessibilityLabel: "addressFormCity", in: viewController.view) as! InputTextFieldView
+                statePicker = self.findView(withAccessibilityLabel: "addressFormState", in: viewController.view) as! BasePicker
+                zipCodeTextFieldView = self.findView(withAccessibilityLabel: "addressFormZipCode", in: viewController.view) as! InputTextFieldView
+                phoneTextFieldView = self.findView(withAccessibilityLabel: "addressFormPhone", in: viewController.view) as! InputTextFieldView
+                submitButton = self.findView(withAccessibilityLabel: "addressFormSubmitButton", in: viewController.view) as! BlackButton
+            }
+            
+            context("if it doesn't have any symbols in each text field view") {
+                it("needs to disable submit button") {
+                    expect(submitButton.isEnabled) == false
```

Comment:
Add `viewModelMock.makeSubmitButtonDosabled()` before `expect`, it will be more correct, especially in case when this test case will be execute after previous one.